docs/4.4/enterprise/workflow/index.md

```diff
@@ -129,6 +129,87 @@ deny:
     roles: ['admin']
 ```
 
+## Adding a reason to approval requests
+
+Teleport 4.4.4 introduced the ability for users to request additional roles. `tctl`
+or the Access Workflow API makes it easy to dynamically approve or deny these requests.
+
+In the Supporting users that start in an essentially unprivileged state and must
+always go through the dynamic access API in order to gain meaningful privilege.
+
+Leveraging the claims (traits) provided by external identity providers both when
+determining which roles a user is allowed to request, and if a specific request
+should be approved/denied.
+
+### Example Setup
+
+**Unprivileged User**<br>
+In this example we've a employee, who isn't able to access any systems. When they
+login they'll always need to provide a reason to access.
+
+```yaml
+kind: role
+metadata:
+  name: employee
+spec:
+  allow:
+    request:
+      # the `roles` list can now be a mixture of literals and matchers
```

Comment:
What's a "matcher" and how do we define it? Do we have a list of acceptable values somewhere?

Reply:
I'm pretty sure we only support wildcard matching, such as dev-* ? ( @fspmarshall can you confirm? )

Reply:
Wildcards and regexes, so stuff like this also works:
- `^foo.*$`
- `foo-{{regexp.match("bar")}}-baz`
- `foo-{{regexp.not_match("bar")}}-baz`

Reply:
I've added a quick wildcard/regex tips section.